### PR TITLE
Add data lake glossary entry

### PR DESCRIPTION
**Pull request by johnblust**
## Description & motivation
Adds the data lake glossary entry to the docs repo

### CODE REVIEW

File: website/docs/terms/data-lake.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+| Snowflake | Snowflake is a fully-managed platform for data warehousing, data lakes, data engineering, data science, and data application development. | Analytics engineers, Data analysts,  Business analysts |
+
+## Conclusion
+
+The data lake is the younger data management platform compared to its data warehouse counterpart. Because of its unique ability to hold large amounts of data in its native, raw format, it has allowed organizations to store all its data in a centralized place, even if sometimes they don’t have a definitive use case for the data yet. In addition, it serves as a great buffer and landing zone for data before it is ultimately transformed and aggregated in a data warehouse. Lastly, it has unlocked a world of new possibilities by enabling organizations to build data science and machine learning use cases on top of it. The data lake is an integral pillar in the Modern Data Stack and the practice of analytics engineering.
+
+## Additional reading
+- [Glossary: Data warehouse](https://docs.getdbt.com/terms/data-warehouse)
```
Comment (runleonarun) [inline website/docs/terms/data-lake.md]: @johnblust and @KiraFuruichi, apologies about the drive by comment, but I'd love to see us use relative links, especially if we're going to change the domain at some point.
```suggestion
- [Glossary: Data warehouse](/terms/data-warehouse) 
```

File: website/docs/terms/data-lake.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+## What are the drawbacks and challenges when using a data lake? 
+
+For all of the advantages of data lakes listed above such as cost-effectiveness and flexibility, they also come with several drawbacks and challenges.
+
+### Inability to do fast reporting
+
+Query performance and speed is one capability area where data warehouses typically trump data lakes. While structuring data first may seem inflexible and rigid at times, it is the right approach to implement when you have analyses and reports that you want to run frequently. The following are several query performance techniques that can only be applied to data warehouses:
+
+| Performance Technique | Description | Practical scenario |
```
Comment (KiraFuruichi) [inline website/docs/terms/data-lake.md]:
```suggestion
| Performance technique | Description | Practical scenario |
```
consistent capitalization

File: website/docs/terms/data-lake.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+## Data lake vs. data warehouse
+
+|  |  Data lake | Data warehouse |
+|---|---|---|
+| Types of data  | Structured, Semi-Structured, Unstructured | Structured  |
+| Data stored in  | Folders and files in raw format | Schemas and tabular data format |
+| Schema/schema definition | Store data in its raw format, transform the data later | Must know upfront |
+| Intended users | Data engineers, Analytics engineers, Data analysts,  Data scientists | Analytics engineers, Data analysts,  Business analysts |
+| Common use cases | Data archiving and storage, data science and machine learning,exploratory data analysis | Business intelligence,dashboarding,reporting and analytics  |
```
Comment (KiraFuruichi) [inline website/docs/terms/data-lake.md]:
```suggestion
| Common use cases | Data archiving and storage, data science and machine learning, exploratory data analysis | Business intelligence, dashboarding, reporting and analytics  |
```
missing spaces

File: website/docs/terms/data-lake.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+Because you don’t need to impose a formal structure for how data is organized in a data lake, you can perform preliminary data exploration on that data, such as calculate summary statistics, discover anomalies and outliers, and plot data visualizations to derive preliminary insights. Commonly referred to as EDA, this is typically conducted as an initial step before formalizing a data science or machine learning use case. 
+
+## Data lake vs. data warehouse
+
+|  |  Data lake | Data warehouse |
+|---|---|---|
+| Types of data  | Structured, Semi-Structured, Unstructured | Structured  |
+| Data stored in  | Folders and files in raw format | Schemas and tabular data format |
+| Schema/schema definition | Store data in its raw format, transform the data later | Must know upfront |
+| Intended users | Data engineers, Analytics engineers, Data analysts,  Data scientists | Analytics engineers, Data analysts,  Business analysts |
```
Comment (KiraFuruichi) [inline website/docs/terms/data-lake.md]:
```suggestion
| Intended users | Data engineers, analytics engineers, data analysts,  data scientists | Analytics engineers, data analysts,  business analysts |
```

File: website/docs/terms/data-lake.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+| Common use cases | Data archiving and storage, data science and machine learning,exploratory data analysis | Business intelligence,dashboarding,reporting and analytics  |
+
+## Data platforms that support data lake workloads
+
+|  Data Platform | Description | Data warehouse |
+|:---:|:---:|---|
+| Cloudera | Cloudera Open Data Lakehouse is a platform that provides data lake flexibility and data warehouse performance in a single platform.  | Structured  |
+| Databricks | Databricks is a cloud-based collaborative data science, data engineering, and data analytics platform that brings the best of data warehouses and data lakes into a single unified platform.  | Schemas and tabular data format |
+| Dremio | Dremio is the data lakehouse platform built for SQL and built on open source technologies that both data engineers and data analysts love. Dremio powers BI dashboards and analytics directly on data lake storage. | Must know upfront |
+| Snowflake | Snowflake is a fully-managed platform for data warehousing, data lakes, data engineering, data science, and data application development. | Analytics engineers, Data analysts,  Business analysts |
```
Comment (KiraFuruichi) [inline website/docs/terms/data-lake.md]:
```suggestion
| Snowflake | Snowflake is a fully-managed platform for data warehousing, data lakes, data engineering, data science, and data application development. | Analytics engineers, data analysts,  business analysts |
```

**Review by KiraFuruichi — APPROVED** [review]
Left some minor comments that should be merged in (and +1 to Leona's comment). Thx for getting this up and gonna give it the green light, but please merge in the minor changes!

**Comment by johnblust** [comment]
Thanks for the review and edits everyone! Will merge in now

**Pull request merged**